### DISCUSSION TITLE
Need help with Push-Notifications on iOS/Android

### DISCUSSION THREAD

**Discussion by xKugeki** [dani-garcia/vaultwarden#3667] [Q&A] [answered]
Hello everyone :)

i recently enabled weebhook and figured out that this feature is awesome :)
now i wanted to enable push [(3304)](https://github.com/dani-garcia/vaultwarden/pull/3304) so that I have the same feature for my mobile devices.

I edited my env:
![image](https://github.com/dani-garcia/vaultwarden/assets/119607113/05c83a6d-15d4-4558-bf96-489a63c07131)

but my entries i edit @ my web-vault or desktop-app are not synced with the iOS App.
Just manual sync will show my change. Even after waiting a couple of minutes.

I allready tried stuff from [(3577)](https://github.com/dani-garcia/vaultwarden/issues/3577) and reinstalled the iOS App and allready deauthorized all sessions.
After Reinstalling the App i got the Notification that this App want to use Push Notifications and allowed it.

Is there maybe another port that i had to forward or stuff i have to edit in npm?
Right now it is 8083, 80, 443 and 3012 i allowed on my firewall
(i use NPM as Reverse Proxy)

Thanks in advance :)

**Comment by BlackDex** [comment]
In the Vaultwarden database there should be a value in the `push_token` column for this to work.
Without that, the mobile client did not yet fully or correctly registered for this feature to work.

Without this data we can't send push messages.
You could try to enable `Approve login requests` feature in the mobile client, and after that click on the `Pending login requests` button, try to do a manual sync, logout, and login again to see if that somehow triggers it.

I actually have no clue as of yet if there is a way to force this register from the mobile if we even can.

**Reply by xKugeki** [reply]
mhm I deauthorized all sessions then the tables in 'devices' gets deleted.
After Reinstalling the App, Login with 2FA, Enable login request and manual syncing the 'push:_token' stays empty.

When I click on pending login requests I get an error message.

Sad :/

**Reply by BlackDex** [reply]
The error because of pending login request is normal, we do not support that yet, but i kinda hoped it might trigger something.

**Reply by xKugeki** [reply]
When i check my config.json there is nothing in it like 'push_enabled' or the Key and the ID?
Could this maybe be a problem aswell?

**Reply by xKugeki** [reply]
Getting this error in the logs when i try to click on pending login requests:
![image](https://github.com/dani-garcia/vaultwarden/assets/119607113/8025fdcb-20a7-487a-b8f0-c831b0cd884e)

**Reply by SoOutnumb3r3d** [reply]
I too had trouble getting my iOS client re-registered for push notifications after de-authorizing.  Turns out I just needed patience!  I saw here that the clients register for push once a day: https://contributing.bitwarden.com/architecture/deep-dives/push-notifications/mobile.  I found that it had successfully re-registered itself for push shortly after midnight when I was experimenting with it.

**Reply by xKugeki** [reply]
@SoOutnumb3r3d thanks for your answer :D i will give it a try and let you know ^^

**Reply by BlackDex** [reply]
Those values not in the `config.json` shouldn't be an issue. As long as you can see them in the admin interface under the read-only settings.
If they are there, they are active.

Other then trying to get the device to correctly register it self, I do not know what else we can do at the moment.

**Reply by xKugeki** [reply]
@BlackDex for some reason i cant see them inside the read only config :/ i allready rebuild the container in portainer

**Reply by BlackDex** [reply]
You do not see those config items there, or you do not see the values there, but do see the config items?

Because, if you do not see the config items, then i doubt you are using the `:testing` tagged image. Or Portainer (as i see often) didn't started the newer version.

**Reply by xKugeki** [reply]
@BlackDex 
I can see the following in the Read-Only Config:
![image](https://github.com/dani-garcia/vaultwarden/assets/119607113/9486d65f-a9ef-40bb-98bf-63c798ac07f6)
![image](https://github.com/dani-garcia/vaultwarden/assets/119607113/fb5c1d9b-788c-4f80-9585-cdd1c5fba6ec)
![image](https://github.com/dani-garcia/vaultwarden/assets/119607113/a92960ec-8c86-498a-921e-6c0d90654fe8)

Im using the **vaultwarden/server:latest** Image. Should i go with **vaultwarden/server:1.28.1** instead? And yeah im using Portainer ^^

Edit: Recreating with **vaultwarden/server:1.28.1** brings the same results

**Reply by SoOutnumb3r3d** [reply]
At the moment you need to be using vaultwarden/server:testing for mobile push notifications to work.

**Comment by BlackDex** [comment, marked as answer]
You need to use `:testing` not `:latest`.
`:latest` doesn't support push and web-sockets via port 80.

**Reply by xKugeki** [reply]
oh ^^'
well that makes everything clear now :D 
thank you :D now i see the entries and the push_token is generated :)
![image](https://github.com/dani-garcia/vaultwarden/assets/119607113/4431cf73-14ea-45ee-93ed-04a66ccb71f8)

Just noticed that the personal vault entries get synced in realtime but the organisation vault entries wont get synced - just after manual sync ^^ 

But thank you very much for you help and for your time, I really appreciate that.

**Reply by BlackDex** [reply]
Org items are not synced via push. That isn't something supported by Bitwarden if I'm correct.

**Reply by xKugeki** [reply]
oh okay good to know :)
I´ll close the topic :)
